jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/Query.java

```diff
@@ -18,16 +18,27 @@
 
     public abstract Optional<List<GetAccountTransactionBySequenceNumber>> getAccountTransactionBySequenceNumberQueries();
 
+    public abstract Optional<List<GetTransactions>> getTransactions();
+
     public List<RequestItem> toGrpcObject() {
-        return Stream.concat(
-                getAccountStateQueries()
-                        .map(Collection::stream)
-                        .orElse(Stream.empty())
-                        .map(GetAccountState::toGrpcObject),
-                getAccountTransactionBySequenceNumberQueries()
-                        .map(Collection::stream)
-                        .orElse(Stream.empty())
-                        .map(GetAccountTransactionBySequenceNumber::toGrpcObject))
-                .collect(toList());
+        Stream<RequestItem> getAccountStateStream = getAccountStateQueries()
+                .map(Collection::stream)
+                .orElse(Stream.empty())
+                .map(GetAccountState::toGrpcObject);
+        Stream<RequestItem> getAccountTransactionsStream =getAccountTransactionBySequenceNumberQueries()
+                .map(Collection::stream)
+                .orElse(Stream.empty())
+                .map(GetAccountTransactionBySequenceNumber::toGrpcObject);
+        Stream<RequestItem> getTransactions = getTransactions()
+                .map(Collection::stream)
+                .orElse(Stream.empty())
+                .map(GetTransactions::toGrpcObject);
+
+        Stream<RequestItem> totalStream = Stream.concat(
+                getAccountStateStream,
+                getAccountTransactionsStream);
+        totalStream= Stream.concat(totalStream, getTransactions);
+
+        return totalStream.collect(toList());
     }
```

Comment:
Sure,In some scenarios, we can get some transaction events by pagination.I think this is the same purpose as the API provided by grpc . Regarding Stream.concat is really not suitable , implement the getTransactions example and did not optimize this part of the original sample.

Reply:
What about using `Stream.builder()` instead of concat? It's a bit cleaner in my opinion. If there are more than 2 streams to append together `concat()` tends to look messy. My suggestion:
```
    public List<RequestItem> toGrpcObject() {

        Stream.Builder<RequestItem> resultBuiler = Stream.builder();

        getAccountStateQueries()
                .map(Collection::stream)
                .orElse(Stream.empty())
                .map(GetAccountState::toGrpcObject)
                .flatMap(Collection::stream)
                .forEach(resultBuiler);

        getAccountTransactionBySequenceNumberQueries()
                .map(Collection::stream)
                .orElse(Stream.empty())
                .map(GetAccountTransactionBySequenceNumber::toGrpcObject)
                .flatMap(Collection::stream)
                .forEach(resultBuiler);

        getTransactions()
                .map(Collection::stream)
                .orElse(Stream.empty())
                .map(GetTransactions::toGrpcObject)
                .flatMap(Collection::stream)
                .forEach(resultBuiler);

        return resultBuiler.build()
                .collect(toList());
    }
```

Reply:
This is a pretty good optimization。

Reply:
Adjusted accordingly